main.ts

```diff
@@ -1,135 +1,110 @@
-import { MarkdownView, Plugin, EditorPosition, Editor } from "obsidian";
-
+import { COMMANDS } from "const";
+import { Editor, EditorPosition, Plugin } from "obsidian";
+
+// Override the obsidian module by adding the non-documented Editor methods
+// This makes it so that TypeScript stops shouting at you for accessing a property that it thinks doesn't exist
+// And gives you type-safety

```

Comment:
so it doesn't mark things like `wordAt` and `head`? ok that's good. No idea how this overriding works, but useful :P

Reply:
Haha I also don't fully understand how it works.
But any time you use a non-exposed part of the Obsidian API, you probably want to override it like this for the type-safety